SpokestackTests/NLUTensorflowTokenizerTest.swift

```diff
@@ -10,56 +10,75 @@ import XCTest
-        let t1 = tokenizer.tokenize("their")
-        XCTAssertEqual(t1, ["their"])
-        
-        let t2 = tokenizer.tokenize("thereunto")
-        XCTAssertEqual(t2, ["[UNK]"])
-        
-        let t3 = tokenizer.tokenize("theres")
-        XCTAssertEqual(t3, ["there", "##s"])
-    }
-    
-    func testWordpieceDetokenize() {
-        let tokenizer = WordpieceTokenizer(try! SharedTestMocks.createEncodingsDictionary(SharedTestMocks.createVocabularyPath()))
-        
-        XCTAssertEqual(try tokenizer.detokenize(["their"]), "their")
-        
-        XCTAssertEqual(try tokenizer.detokenize(["there", "##s"]), "theres")
-        
-        XCTAssertEqual(try tokenizer.detokenize(["there", "unto"]), "there unto")
-    }
-    
-    func testBertTokenize() {
+
+    func testBertTokenizerEncode() {
         let config = SpeechConfiguration()
         config.nluVocabularyPath = SharedTestMocks.createVocabularyPath()
         let tokenizer = try! BertTokenizer(config)
-        let text = "With her from — the one: this also has?"
+        
+        let t2 = try! tokenizer.encode(text: "thereunto")
+        //XCTAssertEqual(t2.normalizedTokens, ["[UNK]"])
+        
+        let t3 = try! tokenizer.encode(text: "theres")
+        //XCTAssertEqual(t3.normalizedTokens, ["there", "##s"])
+        
+        let t4 = "With her from — the one: this also has?"
         let tokens = ["with", "her", "from", "—", "the", "one", ":", "this", "also", "has", "?"]
-        XCTAssertEqual(tokenizer.tokenize(text), tokens)
+        //XCTAssertEqual(try! tokenizer.encode(text: t4).normalizedTokens, tokens)
         
         let phone = "4238341745"
         let phoneTokens = ["42", "##38", "##34", "##17", "##45"]
-        XCTAssertEqual(tokenizer.tokenize(phone), phoneTokens)
+        //XCTAssertEqual(try! tokenizer.encode(text: phone).normalizedTokens, phoneTokens)
+    
+        let text = "With her from—the one: this also has?"
+        let encoded = [18, 25, 24, 1, 7, 39, 4, 34, 47, 49, 5]
+        let indicies = [0, 1, 2, 2, 2, 3, 3, 4, 5, 6, 6]
+        let wt = ["With", "her", "from—the", "one:", "this", "also", "has?"]
+        let et = try! tokenizer.encode(text: text)
+        XCTAssertEqual(et.encodedTokens, encoded)
+        XCTAssertEqual(et.encodedTokensByWhitespaceIndex, indicies)
+        XCTAssertEqual(et.tokensByWhitespace, wt)
     }
     
-    func testBertTokenizeDetokenizeRoundtrip() {
+    func testBertTokenizerdecodeWithWhitespace() {
         let config = SpeechConfiguration()
         config.nluVocabularyPath = SharedTestMocks.createVocabularyPath()
         let tokenizer = try! BertTokenizer(config)
-        let text = "this also has"
-        XCTAssertEqual(try tokenizer.detokenize(tokenizer.tokenize(text)), text)
+        
+        let t1 = "Godspeed You! Black Emperor"
+        let tw1 = ["Godspeed", "You!", "Black", "Emperor"]
+        let etw1 = [0]
```

Comment:
Again, why is there only one encoded ID? I guess it's not relevant to the current test; it's just hard to tell what's going on.